document-store/src/test/java/org/hypertrace/core/documentstore/query/QueryBuilderTest.java

```diff
@@ -0,0 +1,67 @@
+package org.hypertrace.core.documentstore.query;
+
+import static java.util.Collections.emptyList;
+import static org.hypertrace.core.documentstore.expression.operators.SortOrder.DESC;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.hypertrace.core.documentstore.expression.impl.ConstantExpression;
+import org.hypertrace.core.documentstore.expression.impl.IdentifierExpression;
+import org.hypertrace.core.documentstore.expression.impl.UnnestExpression;
+import org.hypertrace.core.documentstore.query.Query.QueryBuilder;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+
+@TestInstance(Lifecycle.PER_CLASS)
+class QueryBuilderTest {
+
+  private QueryBuilder queryBuilder;
+
+  @BeforeEach
+  void setUp() {
+    queryBuilder = Query.builder();
+  }
+
+  @Test
+  void testSetAndClearSelections() {
+    final Query queryWithSelections =
+        queryBuilder.addSelection(ConstantExpression.of("Something")).build();
+    assertEquals(1, queryWithSelections.getSelections().size());
+
+    final Query queryWithoutSelections = queryBuilder.setSelections(emptyList()).build();
+    assertEquals(0, queryWithoutSelections.getSelections().size());
```

Comment:
So, previously, this was failing, right?

Reply:
That's correct.